dvc/remote/base.py

```diff
@@ -184,6 +200,10 @@ def _collect_dir(self, path_info):
             for checksum in checksums:
                 entry = dir_info[checksum]
                 entry[self.PARAM_CHECKSUM] = checksum.result()
+        return dir_info
+
+    def _collect_dir(self, path_info):
+        dir_info = self._get_dir_info(path_info)
```

Comment:
I don't see a point in splitting this into two functions, especially because the next sorting doesn't make much sense separately.